exceptiongroup/_tools.py

```diff
@@ -6,6 +6,25 @@
 from . import ExceptionGroup
 
 
+def leaf_exceptions(exc):
+    """ Iterate and extract sub exceptions from given exception.
+
+    Args:
+        exc (BaseException): Exception object we want to extract.
+    Yields:
+        leaf exception object.
```

Comment:
Usually we would write this as "Returns: an iterable of [...]"; to me "Yields" seems to be tied to a specific implementation (a generator) which we might not choose to keep in the future.